test.js

```diff
@@ -562,7 +563,7 @@ test('server timeout', async function (t) {
   await sub
 
   req.end()
-  server.close()
+  setImmediate(() => server.close())
```

Comment:
```suggestion
  setImmediate(() => server.close())
```

I imagine the need for `setImmediate()` is due to the immediate closure of _all_ sockets on `server.close()` so that shouldn't be necessary once only idle connections are closed.

Reply:
That is still a missing piece, after the `idle` first adjustments from https://github.com/holepunchto/bare-http1/pull/15/commits/7b27037279ad23baf14e3075a72356c1715caa42 and https://github.com/holepunchto/bare-http1/pull/15/commits/3aa152057c018f2a1234f4ceee73c7803cedaaed, `setImmediate` is preventing several tests from locking the process after the tests end, and it is needed for more cases, probably a race condition somewhere.